weave/weave_client.py

```diff
@@ -457,9 +459,21 @@ def create_call(
         op: Union[str, Op],
         parent: Optional[Call],
         inputs: dict,
-        attributes: dict = {},
+        attributes: Optional[dict] = None,
         display_name: Optional[str] = None,
+        *,
+        _use_stack: bool = True,
     ) -> Call:
+        """Create, log, and push a call onto the runtime stack.
+
+        Args:
+            op (Union[str, Op]): The operation to log.
+            parent (Optional[Call]): The parent call. If parent is not provided, the current run is used as the parent.
+            inputs (dict): The inputs to the operation.
+            display_name (Optional[str], optional): The display name for the call. Defaults to None.
+            attributes (Optional[dict], optional): The attributes for the call. Defaults to None.
+            _use_stack (bool, optional): Whether to push the call onto the runtime stack. Defaults to True.
```

Comment:
Why prefix with "_"? I think its just as obscure as "parent". I'd just call it "use_stack".

I think the docstring could cover both cases of stack use: If True, read parent from call_context, and push call onto call_context.

Should parent and use_stack be mutually exclusive? I mean should we throw an error if both are passed?

Ideally parent would have a default value of None and be moved to after inputs. This would mean we need to update positional callers, but that's probably not too bad right now since they're all in our codebase.

Reply:
removed underscore. I like to use that when the param is not something we want to be used externally and therefore have more liberties to change

Reply:
we don't want mutual exclusivity. We want to be able to specify a parent and to not use the stack (this is the exact case in Langchain integration)

Reply:
moved parent to 3rd